### ISSUE TITLE
[java] UnusedModifier doesn't check annotations inner classes

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#247]
**Rule Set:** [unused/UnusedModifier](https://pmd.github.io/pmd-5.5.3/pmd-java/rules/java/unusedcode.html#UnusedModifier)

**Description:**
Nested classes in annotations are `public static` by default, no other modifiers are allowed. Explicitly setting them matches this rule's criteria:

> For historical reasons, modifiers which are implied by the context are accepted by the compiler, but are superfluous.

The same applies to inner interfaces and annotations on annotations, which are `public` by default.

**Code Sample demonstrating the issue:**

```java
public @interface Test {
  public static Inner { }
  public interface Interface { }
  public @interface InnerAnnotation { }
}
```

PMD will find no violations on this annotation, 4 violations are expected.

**Running PMD through:** Any

Relate to #246
